tests/spec_test_runners/bls/mod.rs

```diff
@@ -1,47 +1,174 @@
 use crate::test_utils::{load_yaml, TestCase};
 use ethereum_consensus::crypto::{
-    // aggregate, aggregate_verify, fast_aggregate_verify, PublicKey,
-    SecretKey,
-    Signature,
+    aggregate, aggregate_verify, fast_aggregate_verify, PublicKey, SecretKey, Signature,
 };
-// use ethereum_consensus::primitives::Bytes32;
+use ethereum_consensus::primitives::Bytes32;
 use ethereum_consensus::serde as eth_serde;
 use serde::Deserialize;
 use serde_with::{serde_as, DefaultOnError};
 
-pub struct AggregateHandler;
+#[derive(Debug, Deserialize)]
+struct AggregateTestCase {
+    input: Vec<Signature>,
+    output: Option<Signature>,
+}
+
+#[derive(Debug)]
+pub struct AggregateHandler {
+    test_case: AggregateTestCase,
+}
 
 impl AggregateHandler {
     pub fn from(test_case_path: &str) -> Self {
-        Self
+        let path = test_case_path.to_string() + "/data.yaml";
+        let test_case: AggregateTestCase = load_yaml(&path);
+        Self { test_case }
```

Comment:
we may also be able to make a default impl here to skip having to define all of this code for each thing